packages/frontend/src/features/sqlRunner/components/Tables.tsx

```diff
@@ -1,26 +1,107 @@
 import {
     ActionIcon,
+    Box,
     Center,
+    CopyButton,
     Highlight,
     Loader,
     ScrollArea,
     Stack,
     Text,
     TextInput,
+    Tooltip,
     UnstyledButton,
 } from '@mantine/core';
-import { useDebouncedValue } from '@mantine/hooks';
-import { IconSearch, IconX } from '@tabler/icons-react';
-import { useState, type FC } from 'react';
+import { useDebouncedValue, useHover } from '@mantine/hooks';
+import { IconCopy, IconSearch, IconX } from '@tabler/icons-react';
+import { memo, useState, type FC } from 'react';
 import MantineIcon from '../../../components/common/MantineIcon';
+import { useIsTruncated } from '../../../hooks/useIsTruncated';
 import { useTables } from '../hooks/useTables';
 import { useAppDispatch, useAppSelector } from '../store/hooks';
 import { toggleActiveTable } from '../store/sqlRunnerSlice';
 
+const TableItem: FC<{
+    table: string;
+    search: string;
+    schema: string;
+    database: string;
+    isActive: boolean;
+}> = memo(({ table, search, schema, database, isActive }) => {
+    const { ref: hoverRef, hovered } = useHover();
+    const { ref: truncatedRef, isTruncated } = useIsTruncated<HTMLDivElement>();
+    const dispatch = useAppDispatch();
+
+    return (
+        <Box ref={hoverRef} pos="relative">
+            <UnstyledButton
+                onClick={() => {
+                    dispatch(toggleActiveTable(table));
+                }}
+                fw={500}
+                w="100%"
+                p={4}
+                fz={13}
+                c={isActive ? 'gray.8' : 'gray.7'}
+                bg={isActive ? 'gray.1' : 'transparent'}
```

Comment:
Since you already have an sx on this, maybe put these active styles in there? It's kind of a small thing but I think it's cleaner to have one prop for the non-layout styles.

Reply:
On it!

Reply:
https://github.com/lightdash/lightdash/pull/10688/commits/2caae30a883dd3b63137b0ff61b4d94a4898c90e